deltacat/compute/compactor/steps/materialize.py

```diff
@@ -185,23 +214,55 @@ def _materialize(compacted_tables: List[pa.Table]) -> MaterializeResult:
                 f" to download delta locator {delta_locator} with entry ID {src_file_idx_np.item()}"
                 f" is: {download_delta_manifest_entry_time}s"
             )
-            mask_pylist = list(repeat(False, len(pa_table)))
+            record_numbers_tpl_copy = copy.deepcopy(record_numbers_tpl)
```

Comment:
Can we not create a copy? It would use up double the memory than what we had estimated for.